core/src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityIT.java

```diff
@@ -428,6 +430,24 @@ void assertPositionIncrementGapDefaults(String indexName, Version version) throw
         }
     }
 
+    /**
+     * Search on an alias that contains illegal characters that would prevent it from being created after 5.1.0. It should still be
+     * search-able though.
+     */
+    void assertAliasWithBadName(String indexName, Version version) throws Exception {
+        if (version.onOrAfter(Version.fromString("5.1.0-alpha1"))) {
+            return;
+        }
+        // We can read from the alias just like we can read from the index.
+        String aliasName = "#" + indexName;
+        long totalDocs = client().prepareSearch(indexName).setSize(0).get().getHits().totalHits();
+        assertHitCount(client().prepareSearch(aliasName).setSize(0).get(), totalDocs);
```

Comment:
you are asserting that the two hit counts returned are the same. Couldn't they be both `0` in theory?